File: 03_sumOfDiagonals/3.py
from typing import List
class Solution:
    def sumDiagonal(self, N : int , M : List[List[int]] ) -> int :
        diagSum = 0
        for i in range(0, N): 
            diagSum += M[i][i]
            
        return diagSum
    
sol = Solution()
matrix = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]
print("Sum of diagonal:", sol.sumDiagonal(3, matrix))

File: 03_sumOfDiagonals/3.c

#include<stdio.h>

#define N 3

int sumDiagonal(int n, int M[][N]) {
    int sum = 0;
    for (int i=0; i<n; i++){
        sum += M[i][i];
    }
    
    return sum;
}

int main() {
    int matrix[N][N] = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };

    int result = sumDiagonal(N, matrix);
    printf("Sum of main diagonal: %d\n", result);

    return 0;
}
